client/dashboard/domains/domain-connection-setup/components/dns-records-table.tsx

```diff
@@ -0,0 +1,230 @@
+		filterBy: false,
+	},
+];
+
+const fieldsAdvanced: Field< DnsRecordAdvancedVerification >[] = [
+	{
+		id: 'type',
+		label: __( 'Type' ),
+		type: 'text' as const,
+		readOnly: true,
+		enableHiding: false,
+		enableSorting: false,
+		filterBy: false,
+	},
+	{
+		id: 'name',
+		label: __( 'Name' ),
+		type: 'text' as const,
+		readOnly: true,
+		enableHiding: false,
+		enableSorting: false,
+		filterBy: false,
+	},
+	{
+		id: 'currentValue',
+		label: __( 'Current Value' ),
+		type: 'text' as const,
+		readOnly: true,
+		enableHiding: false,
+		enableSorting: false,
+		filterBy: false,
+	},
+	{
+		id: 'expectedValue',
+		label: __( 'Expected Value' ),
+		type: 'text' as const,
+		readOnly: true,
+		enableHiding: false,
+		enableSorting: false,
+		filterBy: false,
+	},
+	{
+		id: 'status',
+		label: __( 'Status' ),
+		type: 'text' as const,
+		readOnly: true,
+		enableHiding: false,
+		enableSorting: false,
+		filterBy: false,
+	},
+];
+
+const VerifiedBadge = () => {
+	return <Badge intent="success">{ __( 'Verified' ) }</Badge>;
+};
+
+const VerifyingBadge = () => {
+	return <Badge intent="warning">{ __( 'Verifying' ) }</Badge>;
+};
+
+const aRecordData = ( currentValue: string, expectedValue: string ) => {
+	return {
+		type: 'A',
+		name: '@',
+		currentValue: currentValue,
+		expectedValue: expectedValue,
+		status: currentValue === expectedValue ? <VerifiedBadge /> : <VerifyingBadge />,
+	};
+};
+
+const wwwCnameRecordData = ( currentValue: string, expectedValue: string ) => {
+	return {
+		type: 'CNAME',
+		name: 'www',
+		currentValue: currentValue,
+		expectedValue: expectedValue,
+		status: currentValue === expectedValue ? <VerifiedBadge /> : <VerifyingBadge />,
+	};
+};
+
+const nameServerRecordData = ( currentValue: string, expectedValue: string ) => {
+	return {
+		currentValue: currentValue,
+		expectedValue: expectedValue,
+		status: currentValue === expectedValue ? <VerifiedBadge /> : <VerifyingBadge />,
+	};
+};
+
+export default function DnsRecordsTable( { domainName }: { domainName: string } ) {
+	const { data: domain } = useSuspenseQuery( domainQuery( domainName ) );
+	const { data: domainMappingStatus } = useSuspenseQuery( domainMappingStatusQuery( domainName ) );
+
+	const isSuggestedMode = domainMappingStatus.mode === DomainConnectionSetupMode.SUGGESTED;
+
+	const dnsRecords = useMemo( (): (
+		| DnsRecordAdvancedVerification
+		| DnsRecordSuggestedVerification
+	)[] => {
+		const data: ( DnsRecordAdvancedVerification | DnsRecordSuggestedVerification )[] = [];
+
+		if ( isSuggestedMode ) {
+			const currentNameServers = ( domainMappingStatus?.name_servers || [] ).sort();
+			const expectedNameServers = [ 'ns1.wordpress.com', 'ns2.wordpress.com', 'ns3.wordpress.com' ];
+			const longestLength = Math.max( currentNameServers.length, expectedNameServers.length );
+
+			for ( let i = 0; i < longestLength; i++ ) {
+				data.push(
+					nameServerRecordData( currentNameServers[ i ] || '-', expectedNameServers[ i ] || '-' )
```

Comment:
It seems weird to fallback to `-` here. Instead, we should render `-` when the value is undefined. WDYT?

Reply:
I'm not sure I follow your suggestion. Do you mean moving the fallback `|| '-'` to inside the `nameServerRecordData` function?

Reply:
Ah I get it now, that makes sense! I've updated the code.